skimage/registration/__init__.py

```diff
@@ -1 +1,8 @@
 from ._optical_flow import optical_flow_tvl1
+from .register_translation import register_translation
+
+
+__all__ = [
+    'optical_flow_tvl1',
+    'register_translation'
+    ]
```

Comment:
We need to think about the API here. `registration.register_translation` no longer makes sense, it would be kind of like `filters.gaussian_filter` rather than `filters.gaussian`.

Some alternatives: `registration.fourier_cross_correlation`, `registration.translation`, `registration.maximum_cross_correlation`, `registration.phase_shift`...

Reply:
I personally vote for `phase_shift`

Reply:
@scikit-image/core anyone else want to weigh in here?

Reply:
Perhaps `phase_correlation`? The original publication describing it is apparently:
```
Kuglin, C., Hines, D., 1975. The phase correlation image alignment method. In: Proc.Internat. Conf. on Cybernetics and Society, IEEE, pp. 163–165
```
and that is also the title of it's [Wikipedia article](https://en.wikipedia.org/wiki/Phase_correlation).

OpenCV has [phaseCorrelate](https://docs.opencv.org/4.2.0/d7/df3/group__imgproc__motion.html#ga552420a2ace9ef3fb053cd630fdb4952) to calculate the shift.

Reply:
There is also the function match_template that would be good to move and rename as well. For this one, it uses a zero normalized cross correlation (https://en.wikipedia.org/wiki/Cross-correlation#Terminology_in_image_processing)

I suggest to think for these two functions at ones, and I would prefer phase_cross_correlation and zero_normalized_cross_correlation.

Reply:
@sciunto, even if `template_matching` uses zncc internally, I don't think that the function name should be modified: template matching operation is well identified. Concerning `phase_cross_correlation` suggestion, why not, but if `phase_correlation` is well accepted in the community, why not keep it?

Reply:
Shouldn't the reference provided by @grlee77 be added to the documentation?

Reply:
template_matching can be used to do other things than matching a template, for instance, you can use it to find shifts between images. Actually, the day I wanted to perform zncc, it took me some time to figure out that template_matching was actually what I wanted to use. That's why I advocate for changing this name.

Reply:
OK @sciunto, that's an interesting point!
The `*_cross_correlation` functions must then coherent behavior. What if the `*_cross_correlation` functions return a `response` images as `match_template` does and make the `register_translation` return a unique `shift` based on any cross correlation method?
The [`_compute_error`](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/register_translation.py#L90) function may then be exposed and the [`return_error`](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/register_translation.py#L134) argument becomes obsolete!

Reply:
That sounds like a good idea. Xref: #3494

Reply:
Do we all agree with `phase_cross_correlation` or do we choose `phase_correlation` instead?

@sciunto, I propose to address #3494 in an other PR ;-)

Reply:
yes, sounds good.